pydantic/main.py

```diff
@@ -324,9 +334,17 @@ def json(
         )
 
     @classmethod
-    def parse_obj(cls: Type['Model'], obj: Mapping[Any, Any]) -> 'Model':
+    def parse_obj(
+        cls: Type['Model'], obj: Optional[Mapping[Any, Any]] = None, __root__: Optional[Any] = None
+    ) -> 'Model':
+
+        if __root__:
+            return cls(__root__=__root__)
+
         if not isinstance(obj, dict):
             try:
+                if obj is None:
```

Comment:
change this to:

```py
if not instanstance(obj, dict):
    if cls._custom_root_type:
        obj = {'__root__': obj}
    else:
        ...
```

Reply:
I think we should also raise an error if the type of `__root__` is a dict.

Reply:
I have changed the condition and added to raise an exception.